trustchain-ion/src/ion.rs

```diff
@@ -0,0 +1,32 @@
+use anyhow::{anyhow, Context, Result};
+use did_ion::sidetree::{is_secp256k1, Sidetree, SidetreeClient, SidetreeError};
+use ssi::jwk::{Algorithm, JWK};
+
+pub const USER_AGENT: &str = concat!(env!("CARGO_PKG_NAME"), "/", env!("CARGO_PKG_VERSION"));
+
+/// Type for ION test network given [did-ion-0.2.0](https://github.com/spruceid/ssi/blob/79ad4a679933d1b4f6af93193408cd0a24e68f74/did-ion/src/lib.rs)
+/// now uses main network.
+#[derive(Clone)]
+pub struct IONTest;
+
+/// did:ion:test Method
+pub type DIDIONTest = SidetreeClient<IONTest>;
+
+impl Sidetree for IONTest {
+    fn generate_key() -> Result<JWK, SidetreeError> {
+        let key = JWK::generate_secp256k1().context("Generate secp256k1 key")?;
+        Ok(key)
+    }
+
+    fn validate_key(key: &JWK) -> Result<(), SidetreeError> {
+        if !is_secp256k1(key) {
+            return Err(anyhow!("Key must be Secp256k1").into());
+        }
+        Ok(())
+    }
+
+    const SIGNATURE_ALGORITHM: Algorithm = Algorithm::ES256K;
+    const METHOD: &'static str = "ion";
+    // Specify "test" network.
+    const NETWORK: Option<&'static str> = Some("test");
```

Comment:
Check that this can't be passed as config?

Reply:
Since this is embedded in the `Sidetree` trait, I'm not sure there is a simple way to set this in the trait impl from config. Perhaps we could just pick which type to use in the [`get_ion_resolver()`](https://github.com/alan-turing-institute/trustchain/blob/bcac76bd7f360250ee6e0b1c59b2678ac790d4ee/trustchain-ion/src/lib.rs#L49) function (and any other place the `ION` type is used)? I've opened  #129 to implement this further as beyond the scope of this PR.

Reply:
agreed, it could be determined by a feature, but not loaded from config

Reply:
Thanks both for the feature suggestion, agreed - I've added this as a task in [#129](https://github.com/alan-turing-institute/trustchain/issues/129#issuecomment-1748841514)